server/game/core/Player.js

```diff
@@ -424,6 +432,21 @@ class Player extends GameObject {
         return null;
     }
 
+
+    /**
+     * Returns ths top cards of the player's deck
+     * @returns {import('./card/CardTypes').PlayableCard[] |null} the Card,© or null if the deck is empty
```

Comment:
If an array return value is expected anyway then it probably makes things more streamlined in some cases on the calling side if we return an empty array instead of null? That way you can do like a for loop over the results without checking